lib/filter-rules-loading.js

```diff
@@ -0,0 +1,146 @@
+        '[MISCONFIGURATION] None of the requested ACCEPT IAC file extensions is compatible',
+      );
+    } else if (!filters.private[0].origin.includes('AZURE')) {
+      // API endpoints for IAC (github, ghe, bitbucket server), doesn't matter for azure, gitlab
+      // file pattern is different for Azure repos, requirements work for all others
+      let template = nestedCopy(
+        filters.private.filter(
+          (entry) =>
+            entry.method == 'GET' &&
+            entry.path.includes('requirements') &&
+            IAC_SCM_ORIGINS.filter((origin) =>
+              entry.origin.includes(`{${origin}}`),
+            ).length > 0,
+        ),
+      );
+      for (let i = 0; i < template.length; i++) {
+        template[i].path = template[i].path
+          .replace('/requirements', '')
+          .replace('%2Frequirements', '');
+        template[i]['//'] = template[i]['//'].replace(
+          'determine the full dependency tree',
+          'scan IAC files',
+        );
+      }
+      for (let i = 0; i < extensions.length; i++) {
+        const extensionTemplate = nestedCopy(template);
+        extensionTemplate[0].path = extensionTemplate[0].path.replace(
+          'txt',
+          extensions[i],
+        );
+        extensionTemplate[1].path = extensionTemplate[1].path.replace(
+          'txt',
+          extensions[i],
+        );
+        filters.private.push(...extensionTemplate);
+      }
+    } else if (filters.private[0].origin.includes('AZURE')) {
+      // Copying and modifying in place in array, not doing NestedCopy here
+      let templateToModify = filters.private.filter(
+        (entry) =>
+          entry.method == 'GET' &&
+          entry.valid &&
+          entry.valid[0].values.includes('**/requirements/*.txt'),
+      );
+      for (let i = 0; i < templateToModify.length; i++) {
+        for (let j = 0; j < extensions.length; j++) {
+          templateToModify[i].valid[0].values.push(`**/*.${extensions[j]}`);
+          templateToModify[i].valid[0].values.push(`**%2F*.${extensions[j]}`);
+        }
+      }
+    } else {
+      logger.error(
+        { accept: process.env.ACCEPT_IAC },
+        'Error Unexpected error at rules injection time. Remove ACCEPT_IAC env var to skip injection logic.',
+      );
+    }
+  }
+  if (process.env.ACCEPT_CODE) {
+    logger.info(
+      { accept: process.env.ACCEPT_CODE },
+      'Injecting Accept rules for Code',
+    );
+    let templateGET = nestedCopy(
+      filters.private.filter(
+        (entry) =>
+          entry.method == 'GET' &&
+          CODE_SCM_ORIGINS.filter((origin) =>
+            entry.origin.includes(`{${origin}}`),
+          ).length > 0,
+      )[0],
+    );
+
+    templateGET.path = '*/info/refs*';
+    templateGET['//'] = 'allow info refs (for git clone)';
+    templateGET.origin = templateGET.origin
+      .replace('https://${GITHUB_TOKEN}', 'https://pat:${GITHUB_TOKEN}')
+      .replace(
+        'https://${GITLAB_TOKEN}@${GITLAB}',
```

Comment:
peeked at the snapshot, looks like gitlab wasn't added. I see ` "origin": "https://\${GITLAB}",` rather than `https://oauth2:${GITLAB_TOKEN}@${GITLAB}'`.

Everything else looks good.

Reply:
My bad, noticed the oauth2 for gitlab but missed actually adding the logic for it. Will fix

Reply:
fixed the gitlab case. Though a bit strange, the gitlab_token env var doesn't seem in use so much till now...